crates/shared/src/oneinch_api.rs

```diff
@@ -384,6 +386,60 @@ where
     serde_json::from_str(&response?).context("1inch result parsing failed")
 }
 
+#[derive(Debug, Clone)]
+pub struct ProtocolCache(Arc<Mutex<TimedSizedCache<(), Vec<String>>>>);
+
+impl ProtocolCache {
+    pub fn new(cache_validity_in_seconds: u64) -> Self {
```

Comment:
Nit: Could be `Duration` and then use .as_secs() to pass into the cache constructor which sadly accepts only integer seconds instead of Duration.